LFS-1142: Create a utility program for changing the version of the CARDS Docker image used in a docker-compose.yml file

IntegralProgrammer opened the pull request:
This PR introduces the `upgrade_cards_image.py` utility program which changes the version of the CARDS Docker image that is specified in a `docker-compose.yml` file.

To test:
- `cd compose-cluster`
- `python3 generate_compose_yaml.py --oak_filesystem`
- `cp docker-compose.yml docker-compose.yml.old`
- `python3 upgrade_cards_image.py --cards_docker_tag 1.0`
- `diff docker-compose.yml.old docker-compose.yml`
- The only change displayed by the `diff` command should be the change of the CARDS Docker image version tag

IntegralProgrammer committed:
LFS-1142: Create a utility program for changing the version of the CARDS Docker image used in a docker-compose.yml file

sdumitriu reviewed (commented):
I feel like using a script to change a compose file is overkill. Why not regenerate the whole compose file?

IntegralProgrammer commented:
> I feel like using a script to change a compose file is overkill. Why not regenerate the whole compose file?

It is possible that we could have a deployment scenario where there are manual modifications done to the `docker-compose.yml` file and automatically regenerating it would not be an option. This script would preserve those manual modifications while upgrading the CARDS Docker image version.

sdumitriu reviewed (commented):
```
upgrade_cards_image.py:32: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
```

Can this warning be addressed?

IntegralProgrammer committed:
LFS-1142: Create a utility program for changing the version of the CARDS Docker image used in a docker-compose.yml file

Specify yaml.SafeLoader as the Loader to be used for loading the source YAML docker-compose file

IntegralProgrammer commented:
> ```
> upgrade_cards_image.py:32: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
> ```
> 
> Can this warning be addressed?

Good observation. I didn't get this deprecation warning on PyYAML version `3.13` (the version provided by the Debian 10 package `python3-yaml`) as it looks like the warning was only added on PyYAML version `5.1+` but I addressed the issue by specifying `Loader=yaml.SafeLoader`.

sdumitriu committed:
LFS-1142: Create a utility program for changing the version of the CARDS Docker image used in a docker-compose.yml file
Make the script executable
